richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -68,15 +68,19 @@ public String getText(Either<L, R> seg) {
     }
 
     @Override
-    public Either<L, R> subSequence(Either<L, R> seg, int start, int end) {
-        return seg.map(l -> lOps.subSequence(l, start, end),
-                       r -> rOps.subSequence(r, start, end));
+    public Optional<Either<L, R>> subSequence(Either<L, R> seg, int start, int end) {
+        return seg.unify(ll -> seg.unify(l -> lOps.subSequence(l, start, end).map(Either::left),
+                                         e -> Optional.empty()),
+                         rr -> seg.unify(e -> Optional.empty(),
+                                         r -> rOps.subSequence(r, start, end).map(Either::right)));
```

Comment:
The `map` method on `Optional` already handles the empty case&mdash;keeps it empty.